mantidproject/mantid
# Tidy up naming of fit workspaces in ads

**Description of work.**

This fixes some usability issues with the new Muon GUI. 
  * A textbox is added which lets a user specify the name of a group in which to store the fit results in the ADS. This name will automatically add an incrementing number after each fit
  * The loading of grouping files is updated so that if a default group pair is present this is detected and set.



**To test:**

  * Load some data EMU19489 which can be found in the unit test directory should do.
  * Check that after loading on the home tab the group/pair combo box is set to long
  * Go to the fitting tab and check that long is automatically selected as the workspace to fit
  * Save and load a grouping xml from the grouping tab and check that the selected group/pair does not change.
  * On the fitting tab do two fits in a row, check that tow groups are created called FittingResults and FittingResults 1. 
  * Change the Workspace Group Name at the bottom of the tab and do a fit check that the appropriate workspace group is added to the ADS



Fixes #25771 #25838  




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
